parla/task_runtime.py

```diff
@@ -976,9 +961,9 @@ class Scheduler(ControllableThread, SchedulerContext):
     # See __init__ function below for comments on the functionality of these members
     _environments: TaskEnvironmentRegistry
     _worker_threads: List[WorkerThread]
+    _free_worker_threads: Deque[WorkerThread]
```

Comment:
`_worker_threads` refers to all of them. `_free_worker_threads` is just the free ones at any given time. Busy threads are popped from this deque, and when they finish their task they reappend themselves to the free deque.